src/orquestra/sdk/_ray/_build_workflow.py

```diff
@@ -343,7 +359,28 @@ def _import_pip_env(ir_invocation: ir.TaskInvocation, wf: ir.WorkflowDef):
             *(task_def.dependency_import_ids or []),
         )
     ]
-    return [chunk for imp in imports for chunk in _pip_string(imp)]
+    sdk_version = version("orquestra-sdk")
+
+    sdk_dependency = None
+    pip_list = [
+        chunk
+        for imp in imports
+        for chunk in _pip_string(imp)
+        if not (sdk_dependency := re.match(r"^orquestra-sdk([<|!|=|>|~].*)?$", chunk))
+    ]
+
+    # If the task definition includes the SDK, warn the user that this does nothing.
+    if sdk_dependency:
+        warnings.warn(
+            f"The definition for task `{ir_invocation.task_id}` "
+            f"declares `{sdk_dependency[0]}` as a dependency. "
+            f"The current SDK version ({sdk_version}) is automatically installed in "
+            "task environments. "
+            "The specified dependency will be ignored.",
+            DeprecationWarning,
+        )
```

Comment:
w.r.t. the warn / fail choice - I have no strong opinion either way. I went with this option because it shortens the time between the user and their results.
w.r.t. transitive dependencies - yep, this is entirely opportunistic. It's easy to give the user an informative warning here so we might as well.

Reply:
I'm happy to go with warn for now, it's an easy change to make it a failure.
Can you change the warning class to a custom warning class? Something like `OrquestraSDKVersionMismatchWarning` or a better name!